src/env.rs

```diff
@@ -277,6 +277,61 @@ mod tests {
         let _zzz = k.open_or_create("zzz").expect("opened");
     }
 
+    #[test]
+    #[should_panic(expected = "wrote: LmdbError(MapFull)")]
+    fn test_exceed_map_size() {
+        let root = Builder::new().prefix("test_exceed_map_size").tempdir().expect("tempdir");
+        println!("Root path: {:?}", root.path());
+        fs::create_dir_all(root.path()).expect("dir created");
+        assert!(root.path().is_dir());
+
+        let k = Rkv::new(root.path()).expect("new succeeded");
+        let sk: Store = k.open_or_create_default().expect("opened");
+
+        // Writing a large enough value should cause LMDB to fail on MapFull.
+        //
+        // The LMDB C library and lmdb Rust crate docs for setting the map size
+        // <http://www.lmdb.tech/doc/group__mdb.html#gaa2506ec8dab3d969b0e609cd82e619e5>
+        // <https://docs.rs/lmdb/0.8.0/lmdb/struct.EnvironmentBuilder.html#method.set_map_size>
+        // both say that the default map size is 10,485,760 bytes, i.e. 10MiB.
+        //
+        // But the DEFAULT_MAPSIZE define in the LMDB code
+        // https://github.com/LMDB/lmdb/blob/26c7df88e44e31623d0802a564f24781acdefde3/libraries/liblmdb/mdb.c#L729
+        // sets the default map size to 1,048,576 bytes, i.e. 1MiB.
+        //
+        // So we write a 1MiB string to the database, which should take a bit
+        // more space than that (for the key, the type, etc.), thus triggering
+        // the failure.
+        let val = "x".repeat(1024 * 1024 /* 1,048,576 bytes, i.e. 1MiB */);
+        let mut writer = k.write().expect("writer");
+        writer.put(&sk, "foo", &Value::Str(&val)).expect("wrote");
+    }
+
+    #[test]
+    fn test_increase_map_size() {
+        let root = Builder::new().prefix("test_open_with_map_size").tempdir().expect("tempdir");
+        println!("Root path: {:?}", root.path());
+        fs::create_dir_all(root.path()).expect("dir created");
+        assert!(root.path().is_dir());
+
+        let mut builder = Rkv::environment_builder();
+        builder.set_map_size(1024 * 1024 * 10 /* 10,485,760 bytes, i.e. 10MiB */);
+        let k = Rkv::from_env(root.path(), builder).unwrap();
+        let sk: Store = k.open_or_create_default().expect("opened");
+        let val = "x".repeat(1024 * 1024 /* 1,048,576 bytes, i.e. 1MiB */);
```

Comment:
You should make sure that this test and the one above use the same value or size (maybe `fn get_larger_than_default_map_size_value()`), and are thus related more closely. Consider a year from now when the default bumps to 2MB, and/or `set_map_size` accrues a bug. The first test will break, and this one will not, because this value is now smaller than the limit.

Reply:
Good point, I've refactored references to the default map size into a get_larger_than_default_map_size_value() function that both tests use to determine a value slightly larger than the default map size.